src/index.js

```diff
@@ -102,59 +109,272 @@ const menu = async () => {
     }
   };
 
+  const battle = async (enemy, player, playerDmg, enemyDmg) => {
+    try {
+      let turn;
+      console.log(`You found ${enemy.name}!\n`);
+      if (player.agi >= enemy.agi) {
+        console.log("You attack first!\n");
+        turn = 1;
+      } else {
+        console.log("The bug attacks first!\n");
+        turn = 0;
+      }
+      while (player.hp > 0 && enemy.hp > 0) {
+        console.log(`${player.name}: ${player.hp}\n${enemy.name}: ${enemy.hp}`);
+        if (turn) {
+          if (playerDmg >= 0) {
+            enemy.hp -= playerDmg;
+            console.log(`You dealt ${playerDmg} dmg to the bug\n`);
+          } else {
+            console.log("Your dmg is null against this bug!\n");
+          }
+          turn--;
+        } else {
+          if (enemyDmg >= 0) {
+            player.hp -= enemyDmg;
+            console.log(`The bug deals ${enemyDmg} dmg to you\n`);
+          } else {
+            console.log(
+              "This bug is so easy to you that it can't deal dmg to you\n"
+            );
+          }
+          turn++;
+        }
+      }
+      if (enemy.hp <= 0) {
+        console.log(`${player.name} wins! Congratulations.\n`);
+      } else if (player.hp <= 0) {
+        console.log(
+          `${enemy.name} wins! Get up and ready to win the next time.\n`
+        );
+      }
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const mission = async (gamer, task, enemies) => {
+    clear();
+    let pDmg,
+      bDmg,
+      enemyList = [1],
+      weak,
+      b4battle = gamer.hp;
+    task.bugs.forEach((enemyId) => {
+      let enemyInfo = enemies.find((bug) => bug.id == enemyId);
+      enemyList.push(enemyInfo);
+    });
+    enemyList.shift();
+    enemyList.forEach(async (foe) => {
```

Comment:
I would like to ask the opinion of other devs about the usage of `forEach` with an async callback, it does not  seem a good idea